awx/main/tasks/jobs.py

```diff
@@ -1066,22 +1057,31 @@ def pre_run_hook(self, job, private_data_dir):
             # ran inside of the event saving code
             update_smart_memberships_for_inventory(job.inventory)
 
+        # Fetch "cached" fact data from prior runs and put on the disk
+        # where ansible expects to find it
+        if job.use_fact_cache:
+            self.facts_write_time = self.instance.start_job_fact_cache(os.path.join(private_data_dir, 'artifacts', str(job.id), 'fact_cache'))
+
     def build_project_dir(self, job, private_data_dir):
         self.sync_and_copy(job.project, private_data_dir, scm_branch=job.scm_branch)
 
-    def final_run_hook(self, job, status, private_data_dir, fact_modification_times):
-        super(RunJob, self).final_run_hook(job, status, private_data_dir, fact_modification_times)
-        if not private_data_dir:
+    def post_run_hook(self, job, status):
```

Comment:
@relrod I wanted to point this out to you because it changes the method interface here, and the way the variables are passed.

In the long run, I don't think there should be any python objects passed from the job preparation into the `post_run_hook` or `final_run_hook`. This still doesn't satisfy that because it saves the last modification time to onto the task (`BaseTask`) instance. My idea for that is that we save a "reference" file and obtain the modification time from that, instead of passing around in-memory.

This is important, because if we ever _de-couple_ job prep & starting from the job wrapup logic, we can't directly pass anything.